File: java/05_021.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
import java.math.*;

public class Main {
    public static void main(String[] args) {
        // Main method code goes here
    }
    
    public void queueImplementations() {
        // When high performance matters
        Deque<String> highPerformanceQueue = new ArrayDeque<>();  // Use this!
        
        // When thread safety matters
        Queue<String> threadSafeQueue = new LinkedBlockingQueue<>();  // Use this!
    }
}

File: data structures and algorithms/cs/05_021.cs
using System;
using System.Collections.Generic;
using System.Collections.Concurrent;

public class MainClass {
    public static void Main(string[] args) {
        // Main method code goes here
    }
    
    public void QueueImplementations() {
        // When high performance matters
        Deque<string> highPerformanceQueue = new LinkedList<string>();  // Use this!
        
        // When thread safety matters
        ConcurrentQueue<string> threadSafeQueue = new ConcurrentQueue<string>();  // Use this!
    }
}
